src/services/TestResultsService.js

```diff
@@ -134,6 +133,9 @@ class TestResultsService {
           })
       }).catch((error) => {
         console.error(error)
+        if (error.statusCode === 400 && error.message === 'The conditional request failed') {
+          return Promise.reject(new HTTPResponse(201, 'Test Result id already exists'))
```

Comment:
Can you add the response in an enum and reuse it below in TestResultsService.unitTest.js?